mne/gui/_coreg_gui.py

```diff
@@ -785,26 +757,48 @@ class CoregPanel(HasPrivateTraits):
     def __init__(self, *args, **kwargs):
         super(CoregPanel, self).__init__(*args, **kwargs)
 
-        # setup save worker
+        # Setup scaling worker
         def worker():
             while True:
-                desc, cmd, args, kwargs = self.queue.get()
-
+                subjects_dir, subject_from, subject_to, scale, bem_names = \
+                    self.queue.get()
                 self.queue_len -= 1
-                self.queue_current = 'Processing: %s' % desc
 
-                # task
+                # Scale MRI files
+                self.queue_current = 'Scaling %s...' % subject_to
                 try:
-                    cmd(*args, **kwargs)
-                except Exception as err:
-                    self.error = str(err)
-                    res = "Error in %s"
+                    scale_mri(subject_from, subject_to, scale, True,
+                              subjects_dir)
+                except:
+                    logger.error('Error scaling %s:\n' % subject_to +
+                                 traceback.format_exc())
+                    self.queue_feedback = ('Error scaling %s (see Terminal)' %
+                                           subject_to)
+                    bem_names = ()  # skip bem solutions
                 else:
-                    res = "Done: %s"
-
-                # finalize
+                    self.queue_feedback = 'Done scaling %s.' % subject_to
+
+                # Precompute BEM solutions
+                for bem_name in bem_names:
+                    self.queue_current = ('Computing %s solution...' %
+                                          bem_name)
+                    try:
+                        bem_file = bem_fname.format(subjects_dir=subjects_dir,
+                                                    subject=subject_to,
+                                                    name=bem_name)
+                        bemsol = make_bem_solution(bem_file)
+                        write_bem_solution(bem_file[:-4] + '-sol.fif', bemsol)
```

Comment:
Replaced the `prepare_bem_model` wrapper with Python code here. Should the wrapper be deprecated?